app/javascript/components/remove-catalog-item-modal.jsx

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Modal } from 'patternfly-react';
+
+const parseApiError = (error) => {
+  const { data: { error: { message } } } = error;
+  return message;
+};
+
+const removeCatalogItems = (catalogItems) => {
+  let apiPromises = [];
+
+  miqSparkleOn();
+  catalogItems.map(item => {
+    apiPromises.push(API.post(`/api/service_templates/${item.id}`, {action: 'delete'}, {skipErrors: [400, 500]})
+                       .then((apiResult) => ({result: 'success', data: apiResult, name: item.name}))
+                       .catch((apiResult) => ({result: 'error', data: apiResult, name: item.name})))
+  });
+  Promise.all(apiPromises)
+    .then((apiData) => {
+      if (catalogItems.length > 1 || (catalogItems.length === 1 && apiData[0].result === 'success')) {
+        window.location.href = '/catalog/explorer?report_deleted=true';
+      } else {
+        add_flash(sprintf(__('Error removing catalog item "%s": %s'), apiData[0].name, parseApiError(apiData[0].data)), 'error');
+        miqSparkleOff();
+      }
+      return apiData;
+    })
+    .then((apiData) => {
+      apiData.map(item => {
+        if (item.result === 'success') {
+          miqFlashLater({message: sprintf(__('The catalog item "%s" has been successfully removed'), item.name)});
+        } else if (item.result === 'error' && catalogItems.length > 1) {
+          miqFlashLater({message: sprintf(__('Error removing catalog item "%s": %s'), item.name, parseApiError(item.data)), level: 'error'});
+        }
+      });
+      miqSparkleOff();
+    });
+};
+
+const isCatalogBundle = (item) => (item.service_type && item.service_type === 'composite');
+
+class RemoveCatalogItemModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: []
+    };
+  }
+
+  componentDidMount() {
+    let apiPromises = [];
+    const catalogItemsIds = ManageIQ.record.recordId ? [ManageIQ.record.recordId] : ManageIQ.gridChecks;
+
+    // Load modal data from API
+    catalogItemsIds.map(item => {
```

Comment:
No need for the `{...}` block for one liners.

Reply:
Fixed.